kymatio/scattering1d/core/scattering1d.py

```diff
@@ -92,7 +96,8 @@ def scattering1d(x, pad, unpad, backend, J, psi1, psi2, phi, pad_left=0,
         # Convolution + downsampling
         j1 = psi1[n1]['j']
 
-        k1 = max(j1 - oversampling, 0)
+        sub1_adj = min(j1, log2_T) if average else j1
```

Comment:
I keep stumbling over these. Maybe a clearer way to formulate things would be
```
k1 = max(j1 - oversampling, 0)
if average:
    k1 = min(k1, kJ)
```
Similar code would work for `k2` below.

The idea being that we want to downsample by `k1`, which is `j1 - oversampling` normally, but if we're averaging, we don't want to downsample by more than the final scale, which is `kJ`.

Reply:
This doesn't scale well. It's more complicated in terms of expanding the definition into its full expression, namely substituting for `kJ`. It's also not `kJ` but `klog_2T`. 

It's fair for Scattering1D, but JTFS uses [more `_adj` logic](https://github.com/OverLordGoldDragon/kymatio/blob/8449dcc6aa620cddfa7ba24e6c3936343bb1d3a7/kymatio/scattering1d/core/timefrequency_scattering.py#L416) that gets out of hand if rewritten this way - not just in more raw code, but in how we think of the logic. Note there isn't even a `k` there, because subsampling is convoluted per the "trimming vs subsampling" distinction; a `kJ` equivalent would be a "higher level" term which obfuscates intent. I would have a much harder time with it.

Then doing `sub1_adj` in time scattering makes it consistent and reading JTFS more familiar.

Reply:
Bottom line, I can do your version for time scattering, but am against reformulating JTFS the same way.